### PR TITLE
Refactor internals and update documentation

### LINKED ISSUE #298: Support agreeSimilarTransaction  for createPayment
**Is your feature request related to a problem? Please describe.**
https://www.paypay.ne.jp/opa/doc/jp/v1.0/direct_debit#operation/createPayment
I want to use `agreeSimilarTransaction` on `createPayment`.

**Describe the solution you'd like**
In php sdk, this parameter is supported. 
https://github.com/paypay/paypayopa-sdk-php/blob/19897d0e8ad3fd2214987d72208cbeec4b726a42/src/Controllers/Payment.php#L46

**Describe alternatives you've considered**
If avoid breaking change, append `createPaymentWithAgreeSimilarTransaction` method.

**Additional context**

### PR DESCRIPTION
**Pull request by curtisfennerpaypay**
This PR refactors the internals of the SDK, while remaining backward compatible.

It also adds some features:

* Fixes #298 by adding an optional boolean parameter the required methods
* Makes all methods return a Promise, making the callback optional, so that code can use await syntax.
* Types have been corrected to be more accurate (particularly the callback/new Promise types)
* Documentation in the README and JsDocs have been updated to be more accurate
* Adds additional test coverage

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing](../blob/master/contributing.md) document?
* [x] Have you read and signed the automated Contributor's License Agreement?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,12 +95,12 @@ let payload = {
     userAgent: "Mozilla/5.0 (iPhone; CPU iPhone OS 10_3 like Mac OS X) AppleWebKit/602.1.50 (KHTML, like Gecko) CriOS/56.0.2924.75 Mobile/14E5239e Safari/602.1"
 };
```
Comment (navin-math) [inline README.md:96]: can we add any info related to agreeSimilarTransaction in read me

**Review by navin-math — CHANGES_REQUESTED** [review]
minor comment

File: README.md (hunk 1 above)
Comment (curtisfennerpaypay) [reply README.md:96]: I've updated the readme with example and explanation.

**Comment by harshitpaypay** [comment]
Looks good to me.

**Review by navin-math — APPROVED** [review]
Approved

**Pull request merged**